src/Sema.zig

```diff
@@ -19514,15 +19492,13 @@ fn structInitAnon(
 
     const types = try sema.arena.alloc(InternPool.Index, extra_data.fields_len);
     const values = try sema.arena.alloc(InternPool.Index, types.len);
-
-    var fields = std.AutoArrayHashMap(InternPool.NullTerminatedString, u32).init(sema.arena);
-    try fields.ensureUnusedCapacity(types.len);
+    const fields = try sema.arena.alloc(InternPool.NullTerminatedString, types.len);
 
     // Find which field forces the expression to be runtime, if any.
     const opt_runtime_index = rs: {
         var runtime_index: ?usize = null;
         var extra_index = extra_end;
-        for (types, 0..) |*field_ty, i_usize| {
+        for (types, values, fields, 0..) |*field_ty, *value, *field, i_usize| {
             const i: u32 = @intCast(i_usize);
```

Comment:
nit: inline this `@intCast` to the only place it's now used, which is in setting `runtime_index`